### DISCUSSION TITLE
when using buf_ring/provider_buffers, what does the len argument mean in io_uring_prep_recv?

### DISCUSSION THREAD

**Discussion by Sherlock-Holo** [axboe/liburing#1180] [Q&A] [answered]
I notice when use buf_ring to provider a buffer for recv op, I can set the `len` as 0, then I notice kernel will try to fill the buffer which provided by buf group

when I set the len > 0, it seems kernel calculate the min len from buf_len and the `len` which I set

however, I didn't see a clear description on https://man.archlinux.org/man/io_uring_prep_recv.3, that make me feel troubled and worried that I may cause problems sometimes

**Comment by axboe** [comment, marked as answer]
len == 0 will always mean "use whatever size the buffer has", and where len != 0 is permitted, it will indeed mean "use the min of len and the buffer length".
